tests/doc_test/doc_dynamic_functions/index.rst

```diff
@@ -37,4 +39,6 @@ DYNAMIC FUNCTIONS
 
         nested id also :need_func:`[[copy("id")]]`
```

Comment:
if `need_func` is deprecated, please add a check for the warning in the test case